software/deploy/deploy.go

```diff
@@ -63,15 +63,15 @@ var tab = printutil.Table{
 	Header:         []string{"#", "LABEL", "DEPLOYMENT NAME", "WORKSPACE", "DEPLOYMENT ID"},
 }
 
-func Airflow(houstonClient houston.ClientInterface, path, deploymentID, wsID, byoRegistryDomain string, ignoreCacheDeploy, byoRegistryEnabled, prompt bool) error {
+func Airflow(houstonClient houston.ClientInterface, path, deploymentID, wsID, byoRegistryDomain string, ignoreCacheDeploy, byoRegistryEnabled, prompt bool) (string, error) {
```

Comment:
I don't understand why we need to return `deploymentID` from here, we aren't updating it in any shape or form as part of this function. So the caller could reuse whatever value it had originally.

Reply:
@neel-astro @kushalmalani 
This function also accepts nil as deploymentID. And inside this function, we prompt the user to enter deploymentID for some cases. That's why this function is also returning the deploymentID on which it operated.
That's why we can't reuse whatever was being passed to this function.